client/app/lib/onboarding.coffee

```diff
@@ -46,11 +54,15 @@ module.exports = class Onboarding
         throw new Error 'Missing mandatory `steps` parameter' unless steps
 
         @user = user
-        @steps = steps.map (step) =>
-            stepModel = new Step step
-            stepModel.onValidated () =>
-                @handleStepSubmitted()
-            return stepModel
+        @steps = steps
+            .reduce (activeSteps, step) =>
+                stepModel = new Step step
+                if stepModel.isActive user
+                    activeSteps.push stepModel
+                    stepModel.onValidated () =>
+                        @handleStepSubmitted()
```

Comment:
For the moment, this method is not designed to be overriden. So let use your solution, thanks 👍